core/capabilities/targets/write_target.go

```diff
@@ -88,10 +86,6 @@ func (cap *EvmWrite) Execute(ctx context.Context, request capabilities.Capabilit
 	cap.lggr.Debugw("Execute", "request", request)
 	// TODO: idempotency
 
-	txm := cap.chain.TxManager()
-
-	config := cap.chain.Config().EVM().ChainWriter()
-
 	reqConfig, err := parseConfig(request.Config)
```

Comment:
There should be a way to pass config into ChainWriter. Under EVM it only contains the destination address, but on solana this could also specify additional metadata that needs to be passed through, e.g. additional accounts to include in the transaction

Reply:
Probably best to do same thing that was done for Chain Reader [link](https://github.com/smartcontractkit/chainlink-common/pull/526). Just add a top level Relayer constructor for Chain Writer.

Reply:
I wish there was a way to provide local configuration though. The values we're setting are specific to keystone/workflows and could potentially be configured by nops to some extent. Having them tweak a toplevel config unrelated to workflows seems unergonomic

Reply:
It should be able to give you a new instance while reusing the chain component so it wouldn't be tweaking anything that is unrelated, so essentially same evm client but different chain writer details.